pkg/client/record/event.go

```diff
@@ -195,10 +202,37 @@ func Event(object runtime.Object, reason, message string) {
 		Count:          1,
 	}
 
-	events.Action(watch.Added, e)
 }
 
 // Eventf is just like Event, but with Sprintf for the message field.
 func Eventf(object runtime.Object, reason, messageFmt string, args ...interface{}) {
 	Event(object, reason, fmt.Sprintf(messageFmt, args...))
 }
+
+type SourceInterface interface {
+	Event(object runtime.Object, reason, message string)
+	Eventf(object runtime.Object, reason, messageFmt string, args ...interface{})
+}
+
+func Source(source *api.EventSource) SourceInterface {
```

Comment:
> I would rather make every call to those functions use FromSource and pass a source.

Not sure about this because it means you can't write a library that could run in multiple binaries. Or I guess you'd have to store the thing FromSource gives you and inject it.